modDbImpl/src/org/aion/db/impl/mockdb/PersistentMockDB.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2017-2018 Aion foundation.
+ *
+ *     This file is part of the aion network project.
+ *
+ *     The aion network project is free software: you can redistribute it
+ *     and/or modify it under the terms of the GNU General Public License
+ *     as published by the Free Software Foundation, either version 3 of
+ *     the License, or any later version.
+ *
+ *     The aion network project is distributed in the hope that it will
+ *     be useful, but WITHOUT ANY WARRANTY; without even the implied
+ *     warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *     See the GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with the aion network project source files.
+ *     If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Contributors:
+ *     Aion foundation.
+ */
+package org.aion.db.impl.mockdb;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import org.aion.base.util.ByteArrayWrapper;
+
+/**
+ * Provides the same behavior as {@link MockDB} with the addition that data is read from a file on
+ * disk at open (if the file exists) and it is stored to disk at close.
+ *
+ * @author Alexandra Roatis
+ */
+public class PersistentMockDB extends MockDB {
+
+    public PersistentMockDB(String name, String path) {
+        super(name);
+
+        Objects.requireNonNull(path, "The database path cannot be null.");
+        this.path = new File(path, name).getAbsolutePath();
+    }
+
+    @Override
+    public boolean open() {
+        if (isOpen()) {
+            return true;
+        }
+
+        LOG.debug("init database {}", this.toString());
+
+        // using a regular map since synchronization is handled through the read-write lock
+        kv = new HashMap<>();
+
+        // load file from disk if it exists
+        File dbFile = new File(path);
+
+        if (dbFile.exists()) {
+            BufferedReader reader = null;
```

Comment:
This little block could be cleaned up a bit using a `try-with-resources` statement